java/src/main/java/com/microsoft/edge/seleniumtools/EdgeOptions.java

```diff
@@ -0,0 +1,317 @@
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package com.microsoft.edge.seleniumtools;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.openqa.selenium.remote.CapabilityType.ACCEPT_INSECURE_CERTS;
+import static org.openqa.selenium.remote.CapabilityType.PAGE_LOAD_STRATEGY;
+import static org.openqa.selenium.remote.CapabilityType.UNEXPECTED_ALERT_BEHAVIOUR;
+import static org.openqa.selenium.remote.CapabilityType.UNHANDLED_PROMPT_BEHAVIOUR;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.Files;
+
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.MutableCapabilities;
+import org.openqa.selenium.PageLoadStrategy;
+import org.openqa.selenium.Proxy;
+import org.openqa.selenium.SessionNotCreatedException;
+import org.openqa.selenium.UnexpectedAlertBehaviour;
+import org.openqa.selenium.remote.BrowserType;
+import org.openqa.selenium.remote.CapabilityType;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+import java.util.stream.Stream;
+
+/**
+ * Class to manage options specific to {@link EdgeDriver}.
+ *
+ * <p>Example usage:
+ * <pre><code>
+ * EdgeOptions options = new EdgeOptions()
+ * options.addExtensions(new File("/path/to/extension.crx"))
+ * options.setBinary(new File("/path/to/msedge"));
+ *
+ * // For use with EdgeDriver:
+ * EdgeDriver driver = new EdgeDriver(options);
+ *
+ * // For use with RemoteWebDriver:
+ * RemoteWebDriver driver = new RemoteWebDriver(
+ *     new URL("http://localhost:4444/wd/hub"),
+ *     new EdgeOptions());
+ * </code></pre>
+ *
+ * @since Since msedgedriver v17.0.963.0
+ */
+public class EdgeOptions extends MutableCapabilities {
+
+    /**
+     * Key used to store a set of EdgeOptions in a {@link Capabilities}
+     * object.
+     */
+    public static final String CAPABILITY = "ms:edgeOptions";
+    public static final String USE_CHROMIUM = "ms:edgeChromium";
+
+    private static final String WEBVIEW_BROWSER_NAME = "webview2";
+
+    private String binary;
+    private List<String> args = new ArrayList<>();
+    private List<File> extensionFiles = new ArrayList<>();
+    private List<String> extensions = new ArrayList<>();
+    private Map<String, Object> experimentalOptions = new HashMap<>();
+
+    public EdgeOptions() {
+        setCapability(CapabilityType.BROWSER_NAME, BrowserType.EDGE);
+        setCapability(USE_CHROMIUM, true);

```

Comment:
Is there an API to set USE_CHROMIUM to false? Do we need a test case for USE_CHROMIUM setup?

Reply:
The Java bindings are Chromium-only unlike the other language bindings. Java users who want to test legacy Edge should continue to use the official Selenium bindings.

This is because of the way the Java EdgeDriver works in Selenium 4 Alpha. They don't have a single class with a useChromium boolean. They have separate EdgeHtmlDriver and EdgeDriver classes. 

So USE_CHROMIUM is always hard-coded to true here.